### PR TITLE
Added support for validator-less bootstrap

### PR DESCRIPTION
**Pull request by Vasu1105**
- Need to upgrade chef version and knife-windows version in gemspec
- Adding spec is pending

### CODE REVIEW

File: lib/chef/knife/azure_server_create.rb [hunk 1]
```diff
@@ -40,7 +41,7 @@ class AzureServerCreate < Knife
 
       def load_winrm_deps
         require 'winrm'
-        require 'em-winrm'
+        # require 'em-winrm'
```
Comment (adamedx) [inline lib/chef/knife/azure_server_create.rb]: Just delete this rather than comment it out.

File: lib/chef/knife/azure_server_create.rb [hunk 2]
```diff
@@ -912,7 +909,26 @@ def get_chef_extension_public_params
...
+          end
+
+          # validator less bootstrap support for bootstrap protocol cloud-api
+          client_builder = Chef::Knife::Bootstrap::ClientBuilder.new(
+            chef_config: Chef::Config,
+            knife_config: config,
+            ui: ui,
+          )
+
+          client_builder.run
```
Comment (siddheshwar-more) [inline lib/chef/knife/azure_server_create.rb]: Do we need to handle any exception for `client_builder.run` ?

File: lib/chef/knife/azure_server_create.rb [hunk 3]
```diff
@@ -912,7 +909,26 @@ def get_chef_extension_public_params
 
       def get_chef_extension_private_params
         pri_config = Hash.new
-        pri_config[:validation_key] = File.read(Chef::Config[:validation_key])
+        if (Chef::Config[:validation_key] && !File.exist?(File.expand_path(Chef::Config[:validation_key])))
```
Comment (siddheshwar-more) [inline lib/chef/knife/azure_server_create.rb]: Put some comments above for why we are doing this?

File: lib/chef/knife/azure_server_create.rb (hunk 2 above)
Comment (Vasu1105) [inline lib/chef/knife/azure_server_create.rb]: There is already a rescue block in parent method create_server_def which calls this block so I don't think there is a need for another rescue block.

**Pull request merged**